backend/danswer/connectors/teams/connector.py

```diff
@@ -0,0 +1,269 @@
+                    )
+                )
+
+            batch_count += 1
+            if batch_count >= self.batch_size:
+                yield doc_batch
+                batch_count = 0
+                doc_batch = []
+        yield doc_batch
+
+    def convert_post_message_list_to_document(
+        self,
+        channel_object: Channel,
+        post_message_list: list[ChatMessage],
+    ) -> Document:
+        most_recent_message_datetime: datetime | None = None
+        semantic_string: str = (
+            "Channel/Post: " + channel_object.properties["displayName"]
+        )
+        post_id: str = channel_object.id
+        web_url: str = channel_object.web_url
+        messages_text = ""
+        post_members_list: list[BasicExpertInfo] = []
+
+        sorted_post_message_list = sorted(
+            post_message_list, key=get_created_datetime, reverse=True
+        )
+
+        if sorted_post_message_list:
+            most_recent_message = sorted_post_message_list[0]
+            most_recent_message_datetime = datetime.strptime(
+                most_recent_message.properties["createdDateTime"],
+                "%Y-%m-%dT%H:%M:%S.%f%z",
+            )
+
+        for message in post_message_list:
+            # add text and a newline
+            if message.body.content:
+                html_parser = HTMLFilter()
+                html_parser.feed(message.body.content)
+                messages_text += html_parser.text + "\n"
+
+            # if it has a subject, that means its the top level post message, so grab its id, url, and subject
+            if message.properties["subject"]:
+                semantic_string += "/" + message.properties["subject"]
+                post_id = message.properties["id"]
+                web_url = message.web_url
+
+            # check to make sure there is a valid display name
+            if message.properties["from"]:
+                if message.properties["from"]["user"]:
+                    if message.properties["from"]["user"]["displayName"]:
+                        message_sender = message.properties["from"]["user"][
+                            "displayName"
+                        ]
+                        # if its not a duplicate, add it to the list
+                        if message_sender not in [
+                            member.display_name for member in post_members_list
+                        ]:
+                            post_members_list.append(
+                                BasicExpertInfo(display_name=message_sender)
+                            )
+
+        # if there are no found post members, grab the members from the parent channel
+        if not post_members_list:
+            post_members_list = self.extract_channel_members(channel_object)
+
+        doc = Document(
+            id=post_id,
+            sections=[Section(link=web_url, text=messages_text)],
+            source=DocumentSource.TEAMS,
+            semantic_identifier=semantic_string,
```

Comment:
You gotta check the Document class. This one is more like Slack, we don't want to index the channel name as a vector/keyword field, it could throw off the search quality significantly.